node-websocket-chat/README.md

```diff
@@ -0,0 +1,200 @@
+
+        export CHAT_ROOT="${PWD}"
+
+7. Update `LD_LIBRARY_PATH`
+
+        export LD_LIBRARY_PATH="${CHAT_ROOT}/idl"
+    
+8. Install node modules.
+
+        npm install
+    
+9. Start the servers in the background 
+
+        node server.js --port 7000 -DCPSConfigFile rtps.ini &
+        node server.js --port 7001 -DCPSConfigFile rtps.ini &
+    
+10. Open `index.html` in a web browser.  Use two tabs or windows to communicate with each server.
+
+    1. In the first tab/window, connect to the first server by using `7000` for the port and clicking "Connect".
+
+       ![Configure the first tab](images/1_first_tab_config.png)
+       
+    2. In the second tab/window, connect to the second server by using `7001` for the port and clicking "Connect".
+
+       ![Configure the second tab](images/2_second_tab_config.png)
+       
+    3. In the first tab/window, type a message.
+
+       ![Type a message in the first tab](images/3_first_tab_type_message.png)
+       
+    4. In the first tab/window, click the "Send Message".
+
+       ![Send a message in the first tab](images/4_first_tab_message_sent.png)
+       
+
+       Check the output from the server.
+       It should be something like
+       
+            Writing message {"message":"A message from the first tab"}
+            Received a message {"message":"A message from the first tab"}
+            Received a message {"message":"A message from the first tab"}
+       
+    5. Switch to the second tab/window to see that it also received the message.
+
+       ![The second tab received the message](images/5_second_tab_message_received.png)
+       
+    6. In the second tab/window, type a message.
+
+       ![Type a message in the second tab](images/6_second_tab_type_message.png)
+       
+    7. In the second tab/window, click the "Send Message".
+
+       ![Send a message in the second tab](images/7_second_tab_message_sent.png)
+       
+    8. Switch to the first tab/window to see that it also received the message.
+
+       ![The first tab received the message](images/8_first_tab_message_received.png)
+
+## Conclusion
+
+Websockets allow clients and servers to efficiently and dynamically exchange messages but create a data distribution problem when multiple server instances are used.
+This article explores the idea of using OpenDDS to distribute data between web server instances using websockets.
+Specifically, the article extends a chat application using OpenDDS with RTPS Discovery so that multiple servers can be used without additional infrastructure.
+The promise of "without additional infrastructure" is a bit misleading since many cloud-based environments don't meet the prerequisites for RTPS Discovery as described [here](https://objectcomputing.com/resources/publications/sett/march-2019-multicast-to-cloud-for-dds-applications) and [here](https://objectcomputing.com/resources/publications/mnb/2019/06/20/interoperable-internet-enabled-dds-applications).
```

Comment:
Maybe rephrase this to focus on the solution -- extending to multiple servers is possible using provided tools.